packages/components/src/internal/popup-utils.ts

```diff
@@ -162,7 +178,10 @@ export const useBoundingRects = (selectors: UseBoundingRectsSelector[]): Boundin
       return { width: 0, height: 0, top: 0, right: 0, bottom: 0, left: 0 };
     });
 
-    if (!_.isEqual(newDimensions, dimensionsRef.current)) {
+    const dimensionsHaveChanged =
+      newDimensions.length !== dimensionsRef.current!.length ||
+      !dimensionsRef.current!.every((d, i) => areBoundingRectsEqual(d, newDimensions[i]));
+    if (dimensionsHaveChanged) {
```

Comment:
I would probably pull in a library for a deep equality check since it took a pretty good amount of code to replace it (and a similar amount of code would likely be necessary everywhere `_.isEqual` is used).